pkg/setup/binary_hyperkube.go

```diff
@@ -9,43 +9,62 @@ import (
 	"fmt"
 	"os"
 	"os/exec"
+	"text/template"
+
+	"github.com/Masterminds/semver"
 
 	"github.com/golang/glog"
 )
 
 func (e *Environment) extractHyperkube() error {
 	glog.V(2).Infof("Extracting %s", e.binaryHyperkube.archivePath)
-	b, err := exec.Command("tar", "-C", e.binABSPath, "-xzvf", e.binaryHyperkube.archivePath, "--strip-components=3", "kubernetes/server/bin/hyperkube").CombinedOutput()
+	b, err := exec.Command("tar", "-C", e.binABSPath, "-xzvf", e.binaryHyperkube.archivePath, "--strip-components=3").CombinedOutput()
 	output := string(b)
 	if err != nil {
 		glog.Errorf("Cannot untar %s, %s: %v", e.binaryHyperkube.archivePath, output, err)
 		_ = e.binaryHyperkube.removeArchive()
 		return err
 	}
-	_, err = os.Stat(e.binaryHyperkube.binaryABSPath)
+
+	// After 1.17 hyperkube binary has been removed from Kubernetes
+	// It's replaced by a shell script forwarding calls to each binary
+	hyperkubeMissingCheck, err := semver.NewConstraint(">=1.17.0-rc")
```

Comment:
can we remove now the `-rc` ?
```suggestion
	hyperkubeMissingCheck, err := semver.NewConstraint(">=1.17.0")
```

Reply:
We can now, updated.